### DISCUSSION TITLE
How can i create a mesh with scroll inside react-three/xr to be scrollable with VR controllers?

### DISCUSSION THREAD

**Discussion by carolinaroqu3** [pmndrs/xr#157] [Q&A] [answered]
I want to create a react component with react-three/xr to be rendered inside VRCanvas that has some elements (in the image example is text but it could just be a list of image meshes in a grid for example) with the ability to be scrollable. Is it possible to create a "scrollable area" inside VRCanvas to be draggable with VR controllers?
![vrscroll](https://user-images.githubusercontent.com/44771728/157030350-fb4ad638-9dd7-43a7-ac2b-57d57ea788b6.png)

**Comment by CodyJasonBennett** [comment]
You'd need to scroll with joysticks using the [gamepad API](https://developer.mozilla.org/en-US/docs/Web/API/Gamepad_API/Using_the_Gamepad_API) while in an immersive session.

**Comment by JenniferFalkenstein** [comment]
Hi there 😄 are there any updates on this one?

**Comment by bbohlender** [comment, marked as answer]
react-three/xr allows scrolling using pointer-events emitted on the scrollable UI. If you use [pmndrs/uikit](https://github.com/pmndrs/uikit), pointer-events allow that scrolling works out of the box.
